mbus-api/src/message_bus/v0.rs

```diff
@@ -168,6 +202,113 @@ pub trait MessageBusTrait: Sized {
         let _ = request.request().await?;
         Ok(())
     }
+
+    /// Get nexuses with filter
+    #[tracing::instrument(level = "debug", err)]
+    async fn get_nexuses(filter: Filter) -> BusResult<Vec<Nexus>> {
+        let nexuses = GetNexuses {
+            filter,
+        }
+        .request()
+        .await?;
+        Ok(nexuses.0)
+    }
+
+    /// Get nexus with filter
+    #[tracing::instrument(level = "debug", err)]
+    async fn get_nexus(filter: Filter) -> BusResult<Nexus> {
+        let nexuses = Self::get_nexuses(filter).await?;
+        only_one!(nexuses)
```

Comment:
The macro usage seems somewhat confusing. The only_one!() macro does 
```rust
if let Some(obj) = list.first() { 
  if list.len() > 1 { } else {} 
}
```
I don't understand why `list.len()` is called since `list.first()` would return none if `list.len()` would be empty?

Reply:
Also when there is no element, i.e `nexuses.empty() == true` should we not return None rather then Error?

Reply:
So the reason was in case there are multiple objects with the same id, and in that case the error "NotUnique" is returned. I was torn between catering for this case or not so I'm happy to remove it.

Reply:
For the empty case, the error was to avoid having an option within the Result.

Reply:
I see. But why is it a Result<T, E> to begin with rather than an Option<T>? Its a bit of bikeshedding perhaps, OTOH, if I have a list get_xx() method having it return what I look for is either there -- or it is not. i.e Some or None. But it seems we at this level need to account for transport errors and that's why it is a Result?

Reply:
> So the reason was in case there are multiple objects with the same id, and in that case the error "NotUnique" is returned. I was torn between catering for this case or not so I'm happy to remove it.

Im not sure what that means... A nexus (should) always be unique?

Reply:
It should be yes, and in k8s it definitely will, was not sure if that would be the case for testing or for other types of objects like pools which have non uuid names. Also if we have ANA I guess we will have >1 nexus with the same UUID? So to get a nexus from a specific node you'd have to use the Filter by Node I guess.

Reply:
> I see. But why is it a Result<T, E> to begin with rather than an Option? Its a bit of bikeshedding perhaps, OTOH, if I have a list get_xx() method having it return what I look for is either there -- or it is not. i.e Some or None. But it seems we at this level need to account for transport errors and that's why it is a Result?

Yes that's right, the Result is to account for transport errors.